src/serviceClients/sqliteClient.ts

```
@@ -85,6 +85,44 @@ export class SQLiteClient {
     }
   }
 
+  public getGpkgTileWidthAndHeight(): { tileWidth: number; tileHeight: number } {
+    let db: SQLiteDB | undefined = undefined;
+    try {
+      db = new Database(this.fullPath, { fileMustExist: true });
+      const sql = `SELECT tile_width,tile_height FROM "gpkg_tile_matrix" group by tile_width,tile_height;`;
+      this.logger.debug({ msg: `Executing query ${sql} on DB ${this.fullPath}` });
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      const tilesSizes = db.prepare(sql).all() as { tile_width: number; tile_height: number }[];
+      if (tilesSizes.length !== 1) {
+        const message = 'invalid gpkg, all tile_width and tile_height must be same pixel size';
+        this.logger.error({
+          tilesSizes: tilesSizes,
+          msg: message,
+        });
+        throw new Error(message);
```

should throw BadRequest as it's invalid GPKG